libraries/rush-lib/src/logic/operations/AsyncOperationQueue.ts

```diff
@@ -177,11 +180,13 @@ export class AsyncOperationQueue
 
   public tryGetRemoteExecutingOperation(): OperationExecutionRecord | undefined {
     const { _queue: queue } = this;
+
     // cycle through the queue to find the next operation that is executed remotely
     for (let i: number = queue.length - 1; i >= 0; i--) {
       const operation: OperationExecutionRecord = queue[i];
 
-      if (operation.status === OperationStatus.RemoteExecuting) {
+      const currentTime: number = new Date().getTime();
+      if (operation.status === OperationStatus.RemoteExecuting && operation.checkAfter < currentTime) {
```

Comment:
Can we make "Remotely executing but a candidate to run locally anyway" into its own status code? Then have the code that sets the "RemoteExecuting" status also own the timeout to flag the operation as available to execute locally if the queue is empty.

Then we can just inject a call to `assignOperations` inside of `onOperationStatusChanged` and not need to do uncomfortable polling inside of the queue itself.

Reply:
Updated! Added a new status to reflect that "remote executing but might be ready for a check" status. Tried to also clean up some of the code that was particular to the remote executing status.